core/src/main/java/io/questdb/griffin/DecimalUtil.java

```diff
@@ -0,0 +1,851 @@
+     * if the type is not supported.
+     */
+    public static int getTypePrecisionScale(int type) {
+        int tag = ColumnType.tagOf(type);
+        if (ColumnType.isDecimalType(tag)) {
+            short p = (short) ColumnType.getDecimalPrecision(type);
+            short s = (short) ColumnType.getDecimalScale(type);
+            return Numbers.encodeLowHighShorts(p, s);
+        }
+        switch (tag) {
+            case ColumnType.DATE:
+            case ColumnType.TIMESTAMP:
+            case ColumnType.LONG:
+                return Numbers.encodeLowHighShorts((short) 19, (short) 0);
+            case ColumnType.INT:
+                return Numbers.encodeLowHighShorts((short) 10, (short) 0);
+            case ColumnType.SHORT:
+                return Numbers.encodeLowHighShorts((short) 5, (short) 0);
+            case ColumnType.BYTE:
+                return Numbers.encodeLowHighShorts((short) 3, (short) 0);
+            default:
+                return 0;
+        }
+    }
+
+    /**
+     * Load any decimal value from a Function into a Decimal256
+     */
+    public static void load(Decimal256 decimal, Function value, @Nullable Record rec) {
+        final int fromType = value.getType();
+        load(decimal, value, rec, fromType);
+    }
+
+    /**
+     * Load any decimal value from a Function into a Decimal256
+     */
+    public static void load(Decimal256 decimal, Function value, @Nullable Record rec, int fromType) {
+        switch (ColumnType.tagOf(fromType)) {
+            case ColumnType.DECIMAL8:
+                byte b = value.getDecimal8(rec);
+                if (b == Decimals.DECIMAL8_NULL) {
+                    decimal.ofNull();
+                } else {
+                    decimal.ofLong(b, ColumnType.getDecimalScale(fromType));
+                }
+                break;
+            case ColumnType.DECIMAL16:
+                short s = value.getDecimal16(rec);
+                if (s == Decimals.DECIMAL16_NULL) {
+                    decimal.ofNull();
+                } else {
+                    decimal.ofLong(s, ColumnType.getDecimalScale(fromType));
+                }
+                break;
+            case ColumnType.DECIMAL32:
+                int i = value.getDecimal32(rec);
+                if (i == Decimals.DECIMAL32_NULL) {
+                    decimal.ofNull();
+                } else {
+                    decimal.ofLong(i, ColumnType.getDecimalScale(fromType));
+                }
+                break;
+            case ColumnType.DECIMAL64:
+                long l = value.getDecimal64(rec);
+                if (l == Decimals.DECIMAL64_NULL) {
+                    decimal.ofNull();
+                } else {
+                    decimal.ofLong(l, ColumnType.getDecimalScale(fromType));
+                }
+                break;
+            case ColumnType.DECIMAL128:
+                long hi = value.getDecimal128Hi(rec);
```

Comment:
profiler highlights this spot as being very inefficient, sum(decimal) is dominated heavily by value retrieval, actual arithmetic pales in comparison